File: app/src/main/java/ca/judacribz/gainzassist/activities/start_workout/SetsAdapter.java
package ca.judacribz.gainzassist.activities.start_workout;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;
import android.widget.TextView;
import java.util.ArrayList;
import java.util.Locale;

import ca.judacribz.gainzassist.R;
import ca.judacribz.gainzassist.models.ExerciseSet;

import static ca.judacribz.gainzassist.util.Calculations.dpToPix;


public class SetsAdapter extends RecyclerView.Adapter<SetsAdapter.SetsViewHolder> {

    private ArrayList<ExerciseSet> exerciseSets;

    // Adapter Constructor
    SetsAdapter(ArrayList<ExerciseSet> exerciseSets) {
        this.exerciseSets = exerciseSets;
    }

    // SetsAdapter @Override
    ///////////////////////////////////////////////////////////////////////////////////////////////
    @NonNull
    @Override
    public SetsViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());

        View view = inflater.inflate(R.layout.part_sets_input, parent, false);
        return new SetsViewHolder(parent.getContext(), view);
    }

    @Override
    public void onBindViewHolder(@NonNull SetsViewHolder holder, int position) {

        ExerciseSet exerciseSet = exerciseSets.get(position);
        holder.bind(exerciseSet.getSetNumber(), exerciseSet.getReps(), exerciseSet.getWeight());
    }

    @Override
    public int getItemCount() {
        return exerciseSets.size();
    }
    ///////////////////////////////////////////////////////////////////////////////////////////////


    // Returns the list of exerciseSets
    public ArrayList<ExerciseSet> getArrayList() {
        return exerciseSets;
    }


    // Custom ViewHolder class for the recyclerView
    // ============================================================================================
    class SetsViewHolder extends RecyclerView.ViewHolder {

        private TextView tvSetNum;
        private EditText etReps, etWeight;
        private Context context;

        // ViewHolder Constructor
        SetsViewHolder(Context context, View itemView) {
            super(itemView);
            this.context = context;

            tvSetNum = itemView.findViewById(R.id.tv_sets);
            etReps = itemView.findViewById(R.id.et_reps);
            etWeight = itemView.findViewById(R.id.et_weight);
        }

        // Sets the text for each list item
        void bind(int setNumber, int reps, float weight) {
            tvSetNum.setText(String.valueOf(setNumber + 1));
            etReps.setText(String.valueOf(reps));
//            etReps.setHeight((int) dpToPix(context, 50f)) ;
            etWeight.setText(String.format(Locale.CANADA, "%.0f", weight));
//            etWeight.setHeight((int) dpToPix(context, 50f)) ;
        }
    }
}

File: app/src/main/java/ca/judacribz/gainzassist/activities/start_workout/SetsAdapter.kt
package ca.judacribz.gainzassist.activities.start_workout

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.EditText
import android.widget.TextView
import ca.judacribz.gainzassist.R
import ca.judacribz.gainzassist.models.ExerciseSet
import java.util.*

class SetsAdapter(private val exerciseSets: ArrayList<ExerciseSet>) :
    RecyclerView.Adapter<SetsAdapter.SetsViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SetsViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        val view = inflater.inflate(R.layout.part_sets_input, parent, false)
        return SetsViewHolder(parent.context, view)
    }

    override fun onBindViewHolder(holder: SetsViewHolder, position: Int) {
        val exerciseSet = exerciseSets[position]
        holder.bind(exerciseSet.setNumber, exerciseSet.reps, exerciseSet.weight)
    }

    override fun getItemCount(): Int {
        return exerciseSets.size
    }

    fun getArrayList(): ArrayList<ExerciseSet> {
        return exerciseSets
    }

    class SetsViewHolder(private val context: Context, itemView: View) :
        RecyclerView.ViewHolder(itemView) {
        private val tvSetNum: TextView = itemView.findViewById(R.id.tv_sets)
        private val etReps: EditText = itemView.findViewById(R.id.et_reps)
        private val etWeight: EditText = itemView.findViewById(R.id.et_weight)

        fun bind(setNumber: Int, reps: Int, weight: Float) {
            tvSetNum.text = (setNumber + 1).toString()
            etReps.setText(reps.toString())
            etWeight.setText(String.format(Locale.CANADA, "%.0f", weight))
        }
    }
}
